src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -666,12 +667,18 @@ private HttpConnection ConstructHttp11Connection(Socket socket, Stream stream, T
         }
 
         // Returns the established stream or an HttpResponseMessage from the proxy indicating failure.
-        private async ValueTask<(Stream, HttpResponseMessage)> EstablishProxyTunnel(CancellationToken cancellationToken)
+        private async ValueTask<(Stream, HttpResponseMessage)> EstablishProxyTunnel(HttpRequestHeaders headers, CancellationToken cancellationToken)
         {
             // Send a CONNECT request to the proxy server to establish a tunnel.
             HttpRequestMessage tunnelRequest = new HttpRequestMessage(HttpMethod.Connect, _proxyUri);
             tunnelRequest.Headers.Host = $"{_host}:{_port}";    // This specifies destination host/port to connect to
 
+            string userAgent = headers.UserAgent.ToString();
```

Comment:
This can be:
```C#
if (headers != null &&
    headers.TryGetValues(HttpKnownHeaderNames.UserAgent, out IEnumerable<string> values))
{
    tunnelRequest.Headers.TryAddWithoutValidation(HttpKnownHeaderNames.UserAgent, values);
}
```
in order to avoid unnecessarily allocating the headers collection and string when there is no user agent explicitly specified.

Reply:
Oh yes, I tried a no alloc way but missed that try method thanks!